test/common/quic/envoy_quic_client_session_test.cc

```diff
@@ -74,22 +77,17 @@ class TestQuicCryptoClientStream : public quic::QuicCryptoClientStream {
   bool encryption_established() const override { return true; }
 };
 
-class TestEnvoyQuicClientSession : public EnvoyQuicClientSession {
+class TestQuicCryptoClientStreamFactory : public EnvoyQuicCryptoClientStreamFactoryInterface {
 public:
-  TestEnvoyQuicClientSession(const quic::QuicConfig& config,
-                             const quic::ParsedQuicVersionVector& supported_versions,
-                             std::unique_ptr<EnvoyQuicClientConnection> connection,
-                             const quic::QuicServerId& server_id,
-                             std::shared_ptr<quic::QuicCryptoClientConfig> crypto_config,
-                             quic::QuicClientPushPromiseIndex* push_promise_index,
-                             Event::Dispatcher& dispatcher, uint32_t send_buffer_limit)
-      : EnvoyQuicClientSession(config, supported_versions, std::move(connection), server_id,
-                               crypto_config, push_promise_index, dispatcher, send_buffer_limit) {}
-
-  std::unique_ptr<quic::QuicCryptoClientStreamBase> CreateQuicCryptoStream() override {
-    return std::make_unique<TestQuicCryptoClientStream>(
-        server_id(), this, crypto_config()->proof_verifier()->CreateDefaultContext(),
-        crypto_config(), this, true);
+  std::unique_ptr<quic::QuicCryptoClientStreamBase>
+  createEnvoyQuicCryptoClientStream(const quic::QuicServerId& server_id, quic::QuicSession* session,
+                                    std::unique_ptr<quic::ProofVerifyContext> verify_context,
+                                    quic::QuicCryptoClientConfig* crypto_config,
+                                    quic::QuicCryptoClientStream::ProofHandler* proof_handler,
+                                    bool has_application_state) override {
+    return std::make_unique<TestQuicCryptoClientStream>(server_id, session,
```

Comment:
should we simply remove this class, and have the mock factory do the custom creation?

Reply:
We don't have mock factories for crypto streams yet, and the mock factories probably will end up doing what these test classes do for now without more interfaces to mock.